pkg/reconciler/ingress/config/gateway.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+const (
+	// GatewayConfigName is the config map name for the gateway configuration.
+	GatewayConfigName = "config-gateway"
+
+	visibilityConfigKey = "visibility"
+
+	// defaultGatewayClass is the gatewayclass name for the gateway.
+	defaultGatewayClass = "istio"
+)
+
+var (
+	// defaultIstioGateway is the default gateway.
+	defaultIstioGateway = &types.NamespacedName{Namespace: "istio-system", Name: "test-gateway"}
```

Comment:
"test-gateway" sounds like it is for testing. not sure what would be the more appropriate name though.